test/integ/test_put_get_medium.py

```diff
@@ -710,49 +757,54 @@ def run(cnx, sql):
 
 @pytest.mark.aws
 @pytest.mark.azure
-def test_put_get_with_hint(tmpdir, conn_cnx, db_parameters):
+@pytest.mark.parametrize("from_stream", [False, pytest.param(True, marks=pytest.mark.skipolddriver)])
+@pytest.mark.parametrize("file_src", ['put_get_1.txt'], indirect=['file_src'])
+def test_put_get_with_hint(tmpdir, conn_cnx, db_parameters, from_stream, file_src):
     """SNOW-15153: PUTs and GETs with hint."""
     tmp_dir = str(tmpdir.mkdir('put_get_with_hint'))
-    data_file = os.path.join(THIS_DIR, "../data", "put_get_1.txt")
+    file_name, file_size, file_stream = file_src
 
     def run(cnx, sql, _is_put_get=None):
-        return cnx.cursor().execute(
-            sql.format(
-                local_dir=tmp_dir.replace('\\', '\\\\'),
-                file=data_file.replace('\\', '\\\\'),
-                name=db_parameters['name']), _is_put_get=_is_put_get).fetchone()
+        sql = sql.format(
+            local_dir=tmp_dir.replace('\\', '\\\\'),
+            file=file_name.replace('\\', '\\\\'),
+            name=db_parameters['name'])
+        if from_stream:
+            return cnx.cursor().execute(sql, _is_put_get=_is_put_get, file_stream=file_stream).fetchone()
+        else:
+            return cnx.cursor().execute(sql, _is_put_get=_is_put_get).fetchone()
```

Comment:
Both if-else clauses could be handled by the helper function that I suggested to be defined more globally